File: src/ipc/canvases.ts
import { invoke } from "@tauri-apps/api/core";
import type { Canvas } from "../types";

export const canvasList = () => invoke<Canvas[]>("canvas_list");
export const canvasCreate = (name: string) => invoke<Canvas>("canvas_create", { name });
export const canvasUpdateViewport = (id: string, x: number, y: number, zoom: number) =>
  invoke<void>("canvas_update_viewport", { id, x, y, zoom });
export const canvasRename = (id: string, name: string) =>
  invoke<void>("canvas_rename", { id, name });
export const canvasDelete = (id: string) => invoke<void>("canvas_delete", { id });

File: src-tauri/src/commands/canvases.rs
use crate::db::{self, canvases::Canvas, Db};
use crate::error::AppResult;
use tauri::State;

#[tauri::command]
pub fn canvas_list(db: State<Db>) -> AppResult<Vec<Canvas>> {
    db::canvases::list(&db)
}

#[tauri::command]
pub fn canvas_create(db: State<Db>, name: String) -> AppResult<Canvas> {
    db::canvases::create(&db, &name)
}

#[tauri::command]
pub fn canvas_update_viewport(
    db: State<Db>,
    id: String,
    x: f64,
    y: f64,
    zoom: f64,
) -> AppResult<()> {
    db::canvases::update_viewport(&db, &id, x, y, zoom)
}

#[tauri::command]
pub fn canvas_rename(db: State<Db>, id: String, name: String) -> AppResult<()> {
    db::canvases::rename(&db, &id, &name)
}

#[tauri::command]
pub fn canvas_delete(db: State<Db>, id: String) -> AppResult<()> {
    db::canvases::delete(&db, &id)
}

#[tauri::command]
pub fn canvas_reorder(db: State<Db>, ids: Vec<String>) -> AppResult<()> {
    db::canvases::reorder(&db, &ids)
}
